src/compute/grid/mod.rs

```diff
@@ -32,19 +32,48 @@ mod track_sizing;
 mod types;
 mod util;
 
+/// The public interface to Taffy's CSS Grid algorithm implementation
+pub(crate) struct CssGridAlgorithm;
+impl LayoutAlgorithm for CssGridAlgorithm {
+    const NAME: &'static str = "CSS GRID";
+
+    fn perform_layout(
+        tree: &mut impl LayoutTree,
+        node: Node,
+        known_dimensions: Size<Option<f32>>,
+        parent_size: Size<Option<f32>>,
+        available_space: Size<AvailableSpace>,
+        _sizing_mode: SizingMode,
+    ) -> SizeAndBaselines {
+        compute_generic(tree, node, known_dimensions, parent_size, available_space, RunMode::PeformLayout)
+    }
+
+    fn measure_size(
+        tree: &mut impl LayoutTree,
+        node: Node,
+        known_dimensions: Size<Option<f32>>,
+        parent_size: Size<Option<f32>>,
+        available_space: Size<AvailableSpace>,
+        _sizing_mode: SizingMode,
+    ) -> Size<f32> {
+        compute_generic(tree, node, known_dimensions, parent_size, available_space, RunMode::ComputeSize).size
+    }
+}
+
 /// Grid layout algorithm
 /// This consists of a few phases:
 ///   - Resolving the explicit grid
 ///   - Placing items (which also resolves the implicit grid)
 ///   - Track (row/column) sizing
 ///   - Alignment & Final item placement
-pub fn compute(
+pub fn compute_generic(
```

Comment:
Can / should this method be factored into the LayoutAlgorithm trait?

Reply:
Well the idea is that the `LayoutAlgorithm` trait has separate `perform_layout` and `measure_size` functions for clarity, but our implementations share one function for both (and my thinking was that it makes sense to keep it that way, but I'm not sure about that). If we don't then we get a bunch of repeated code.

Reply:
Ah got it. I'm fine to keep it as is then.